evm/exceptions.py

```diff
@@ -141,3 +149,25 @@ class OutOfBoundsRead(VMError):
     boundaries of the buffer (such as with RETURNDATACOPY)
     """
     pass
+
+
+class UnannouncedStateAccess(VMError):
+    """
+    Error raised if a storage slot or account field is accessed that has not
+    been specified in the transaction's read or write list, respectively.
+    """
+    pass
+
+
+class GasPriceAlreadySet(Exception):
+    """
+    Error raised if trying to reset the already set gas price.
+    """
+    pass
+
+
+class NotTopLevelCall(Exception):
```

Comment:
@NIC619  ditto

Reply:
The way it's used, I don't believe so.  This isn't something that halts VM execution, but rather to signal to the opcode logic that it's being called incorrectly and to push `0` onto the stack.